pandas/core/arrays/categorical.py

```diff
@@ -2286,42 +2286,24 @@ def isin(self, values) -> npt.NDArray[np.bool_]:
         return algorithms.isin(self.codes, code_values)
 
     def _replace(self, *, to_replace, value, inplace: bool = False):
+        from pandas import (
+            Index,
+            Series,
+        )
+
         inplace = validate_bool_kwarg(inplace, "inplace")
         cat = self if inplace else self.copy()
 
-        # other cases, like if both to_replace and value are list-like or if
-        # to_replace is a dict, are handled separately in NDFrame
-        if not is_list_like(to_replace):
-            to_replace = [to_replace]
-
-        categories = cat.categories.tolist()
-        removals = set()
-        for replace_value in to_replace:
-            if value == replace_value:
-                continue
-            if replace_value not in cat.categories:
-                continue
-            if isna(value):
-                removals.add(replace_value)
-                continue
-
-            index = categories.index(replace_value)
-
-            if value in cat.categories:
-                value_index = categories.index(value)
-                cat._codes[cat._codes == index] = value_index
-                removals.add(replace_value)
-            else:
-                categories[index] = value
-                cat._set_categories(categories)
+        ser = Series(cat.categories, copy=True)
+        ser.replace(to_replace=to_replace, value=value, inplace=True)
```

Comment:
Given that in the future we want to move away from `inplace=True` generally, is there a way to do this without that?

Reply:
Yes, no need to use inplace. I've removed it.

Reply:
@mroeschke gentle ping, any other comments here